src/Lean/Meta/Coe.lean

```diff
@@ -41,10 +41,10 @@ partial def expandCoe (e : Expr) : MetaM Expr :=
       if f.isConst then
         let declName := f.constName!
         if isCoeDecl (← getEnv) declName then
-          for arg in e.getAppArgs do
+          if let some info ← getProjectionFnInfo? declName then
```

Comment:
Is this a fix or an optimization?

Reply:
It doesn't appear to be a *sound* optimization though to me, given that not all `[coe_decl]`s are projections

Reply:
Thanks, could you please summarize that in a comment? :)